### ISSUE TITLE
Panning in a rotated map view with a locked scale changes the magnification factor

### ISSUE THREAD

**Issue by freeshrugsxd** [qgis/QGIS#43275]
**Describe the bug**

When a map view's scale is locked and it has a rotation set, panning the map zooms in.

Also happens with a fresh profile without plugins.

**How to Reproduce**

1. Open New Project
2. Type "world" into coordinate text field in the status bar
3. Zoom in to scale <10.000.000
4. Lock the scale with the padlock icon
5. Set Rotation to 5°
6. Pan the map

**QGIS and OS versions**
QGIS version: 3.18.3-Zürich
QGIS code revision: 735cc85be9
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL/OGR: 3.1.4
Running against GDAL/OGR: 3.1.4
Compiled against GEOS: 3.8.1-CAPI-1.13.3
Running against GEOS: 3.8.1-CAPI-1.13.3
Compiled against SQLite: 3.29.0
Running against SQLite: 3.29.0
PostgreSQL Client Version: 11.5
SpatiaLite Version: 4.3.0
QWT Version: 6.1.3
QScintilla2 Version: 2.10.8
Compiled against PROJ: 6.3.2
Running against PROJ: Rel. 6.3.2, May 1st, 2020
OS Version: Windows 10 (10.0)
Active python plugins: 
AnotherDXF2Shape; 
FOEA; 
GroupStats; 
kbs-group-selected; 
layertilesmapcanvas; 
monokaiprospectrum; 
NNJoin; 
pathfinder; 
pluginbuilder3; 
plugin_reloader; 
QuickOSM; 
SpreadsheetLayers; 
db_manager; 
MetaSearch; 
processing

**Additional context**

Short video showcasing the bug: https://youtu.be/Cis7YtGzczA

**Comment by gioman** [comment]
> without plugins



> Active python plugins:
> AnotherDXF2Shape;
> FOEA;
> GroupStats;
> kbs-group-selected;
> layertilesmapcanvas;
> monokaiprospectrum;
> NNJoin;
> pathfinder;
> pluginbuilder3;
> plugin_reloader;
> QuickOSM;
> SpreadsheetLayers;
> db_manager;
> MetaSearch;
> processing

@freeshrugsxd but you seems to have quite a few active,

**Comment by gioman** [comment]
> @freeshrugsxd but you seems to have quite a few active,

anyway I confirm it also on master/linux.

**Comment by roya0045** [comment]
I'm 105% sure this is a duplicate.

**Comment by gioman** [comment]
> I'm 105% sure this is a duplicate.

Can't find it :)

**Comment by roya0045** [comment]
@gioman #27191 and #38953

**Comment by gioman** [comment]
> @gioman #27191 and #38953

ok but one is closed the other is old and stale (soon to be closed), so this makes sense to stay open for now.

**Comment by gioman** [comment]
> Should I open a new Issue or edit the OP?

edit the title and original description.

**Comment by strk** [comment]
It looks to me that the zoom changes also with rotation set to 0 degrees. Not very noticeable but it seems to grow the image when dropping the cursor near the center of the map and shrink the image when dropping the cursor near any border

**Comment by lukeeastaero** [comment]
Also having trouble with this issue. I believe the magnifier zooms in by rotation*magnification(rounded) on each pan. So if you have 100% and 5 degrees, a pan will zoom into 105%, 111%,117%, etc.

**Comment by freeshrugsxd** [comment]
Hey the issue is still around in 3.20. 
It seems the change in magnification depends heavily on the shape of the map canvas.
Panning a map canvas in landscape or portrait changes the magnifier differently than a square one.

I also noticed that on a wide canvas and 90° rotation, the zoom changes so the new canvas height matches the width of the unrotated canvas. I hope that made sense. I made a short video:
https://youtu.be/Cis7YtGzczA
